lib/oban.ex

```diff
@@ -308,6 +308,8 @@ defmodule Oban do
     insert_all(name, changesets)
   end
 
+  def insert_all(_name, %{changesets: []}), do: []
```

Comment:
Actually, looking at this in the full context I believe it's better to change the base `changesets` clause:

```elixir
  def insert_all(name, %{changesets: changesets}) when is_list(changesets) do
```

```elixir
  def insert_all(name, multi, multi_name, %{changesets: changesets}) when is_list(changesets) do
```

In which case, the tests should stay exactly as you wrote them 🙂

Reply:
Agreed, changed it :)